### PR TITLE
[RFR] Move wall (follows PR#12)

### PR DESCRIPTION
**Pull request by pemoreau**
This corresponds to this [Trello card](https://trello.com/c/ErN0BBnR/2-as-robert-i-can-move-the-walls-of-the-labryrinth)

### CODE REVIEW

File: server/controllers/gameController.js [hunk 1]
```diff
@@ -47,3 +53,30 @@ export const postRotateRemainingPathCard = (req, res) => {
             });
     });
 };
+
+export const postInsertRemainingPathCardAt = (req, res) => {
+    const req_id = req.body.id;
+    const req_x = req.body.x;
+    const req_y = req.body.y;
```
Comment (JulienMattiussi) [inline server/controllers/gameController.js]: destructuring could simplify this
const {id, x, y } = req.body;

File: server/controllers/gameController.js [hunk 2]
```diff
@@ -47,3 +53,30 @@ export const postRotateRemainingPathCard = (req, res) => {
...
+
+    GameModel.findById(req_id, function(err, gameDocument) {
+        if (err) {
+            res.status(400).send('unable to get the game by id');
+            return;
+        }
+
+        console.log('server postInsertRemainingPathCardAt', req_x, req_y);
```
Comment (JulienMattiussi) [inline server/controllers/gameController.js]: log

File: server/controllers/gameController.js [hunk 3]
```diff
@@ -47,3 +53,30 @@ export const postRotateRemainingPathCard = (req, res) => {
...
+        let newGame = insertRemainingPathCardAt(gameDocument.toObject(), req_x, req_y);
+        newGame = toMoveState(newGame);
+
+        gameDocument.set(newGame);
+        gameDocument
+            .save()
+            .then(game => {
+                res.status(200).json(game);
```
Comment (JulienMattiussi) [inline server/controllers/gameController.js]: As all controlleurs using this syntax, You could create en fonction for that and avoid repeating status(200) : status(400) into each

File: src/actions/index.js [hunk 4]
```diff
@@ -41,6 +43,40 @@ export const rotateRemainingPathcardClockwise = game => {
     };
 };
 
+const optimisticInsertRemainingPathcardAt = (dispatch, game, x, y) => {
+    const newGame = insertRemainingPathCardAt(game, x, y);
+    console.log('optimistic new game: ', newGame);
```
Comment (JulienMattiussi) [inline src/actions/index.js]: log

File: src/actions/index.js [hunk 5]
```diff
@@ -41,6 +43,40 @@ export const rotateRemainingPathcardClockwise = game => {
...
+        headers: { 'Content-Type': 'application/json' },
+    };
+
+    return axios(config).then(response => {
+        console.log('network new game: ', response.data);
+        dispatch(insertRemainingPathcardAtSuccess(response.data));
+    });
+};
+
+const convertDisplayToBoard = x => (x === 0 ? -1 : x === 6 ? 7 : x);
```
Comment (JulienMattiussi) [inline src/actions/index.js]: not clear for me
maybe use constant for explaining what is 6, what is 7

File: src/common/game.js [hunk 6]
```diff
@@ -304,13 +304,18 @@ export const insertRemainingPathCard = game => {
 };
 
 export const insertRemainingPathCardAt = (game, x, y) => {
+    console.log('insertRemainingPathCardAt ', { x, y });
```
Comment (JulienMattiussi) [inline src/common/game.js]: log

File: src/common/game.js [hunk 7]
```diff
@@ -304,13 +304,18 @@ export const insertRemainingPathCard = game => {
...
+
     if (!isInsertionPosition({ x, y })) {
+        console.log('not an isInsertionPosition: identity');
```
Comment (JulienMattiussi) [inline src/common/game.js]: log

Comment (JulienMattiussi) [reply src/common/game.js]: and much more next ...

File: server/controllers/gameController.js (hunk 1 above)
Comment (pemoreau) [reply server/controllers/gameController.js]: completely refactored this part in following PR

File: server/controllers/gameController.js (hunk 3 above)
Comment (pemoreau) [reply server/controllers/gameController.js]: Yes. Done in the following PR

**Pull request merged**